### PR TITLE
feat: support engines.node version checker

### PR DESCRIPTION
**Pull request by fengmk2**
closes https://github.com/cnpm/npminstall/issues/156

---
This change is [Reviewable](https://reviewable.io/reviews/cnpm/npminstall/195)

### CODE REVIEW

**Comment by dead-horse** [comment]
readme 里面写一下

File: bin/install.js [hunk 1]
```diff
@@ -92,6 +93,7 @@ Options:
   --trace: show memory and cpu usages traces of installation
   --ignore-scripts: ignore all preinstall / install and postinstall scripts during the installation
   --forbidden-licenses: forbit install packages which used these licenses
+  --engine-strict: refuse to install (or even consider installing) any package that claims to not be compatible with the current Node.js version.
```
Comment (fengmk2) [inline bin/install.js]: @dead-horse 好的，看来得2个地方写。。。

Comment (fengmk2) [inline bin/install.js]: 默认是不开启的，只是 warn

**Comment by fengmk2** [comment]
2.21.0

**Pull request merged**